### ISSUE TITLE
Set a limit to the `DataPlane` number of instances handled by a single `ControlPlane`

### ISSUE THREAD

**Issue by mlavacca** [Kong/kong-operator#188]
## Problem Statement

With Kong/gateway-operator-archive#607, we'll allow the `ControlPlane` to push the configuration to many different `DataPlane` instances. We need to evaluate the number of `DataPlane` instances the `ControlPlane` can handle correctly and either set a hard limit on the `DataPlane` replicas or document the value.

## Additional information

This issue is blocked by 
- [x] Kong/gateway-operator-archive#8 
- [x] Kong/gateway-operator-archive#607 

## Acceptance Criteria
- [ ] Evaluate the number of `DataPlane` instances that the `ControlPlane` can correctly handle
- [ ] validate that the number of replicas is lower or equal to that value, or document the value to provide such information to the user.

**Comment by shaneutt** [comment]
I would suggest if we decide we want the hard limit approach, that we soften that and provide _some_ mechanism to override it. e.g. `--yes-i-know-what-im-doing-and-i-know-this-isnt-supported-just-extend-the-dataplane-limit=10000`

**Comment by pmalek** [comment]
@programmer04 AI: to close this and create a new issue for max DP and CP instances that can be managed by KO.

**Comment by programmer04** [comment]
superseded by https://github.com/Kong/kong-operator/issues/2313
